### PR TITLE
doc: nrf: lib: modem: restructure nrf_modem_lib documentation and add note about RTT tracing

### PR DESCRIPTION
**Pull request by eivindj-nordic**
* The nrf_modem_lib documentation is moved to its own folder and split
into multiple files.

* Add note about RTT tracing to modem trace doc.

### CODE REVIEW

File: doc/nrf/libraries/modem/nrf_modem_lib/index.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+   nrf_modem_lib_pm_integration.rst
+   nrf_modem_lib_trace.rst
+   nrf_modem_lib_fault.rst
+   nrf_modem_lib_api.rst
+
+.. include:: nrf_modem_lib_wrapper.rst
+
+.. include:: nrf_modem_lib_socket_offloading.rst
+
+.. include::  nrf_modem_lib_os_abstraction.rst
```
Comment (trantanen) [inline doc/nrf/libraries/modem/nrf_modem_lib/index.rst:26]:
```suggestion
.. include:: nrf_modem_lib_os_abstraction.rst
```

**Review by trantanen — APPROVED** [review]
Looks good to me

**Comment by lemrey** [comment]
Wasn't there a bot posting the documentation output, what happened to it? :(

**Comment by divipillai** [comment]
> Wasn't there a bot posting the documentation output, what happened to it? :(

May be re-running the CI could generate the preview. I could see the doc preview for other PRs.

**Review by divipillai — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Pull request merged**